staging/src/k8s.io/apiserver/pkg/storage/cacher/cacher.go

```diff
@@ -159,6 +160,49 @@ func (i *indexedWatchers) terminateAll(objectType reflect.Type, done func(*cache
 	}
 }
 
+// As we don't need a high precision here, we keep all watchers timeout within a
+// second in a bucket, and pop up them once at the timeout. To be more specific,
+// if you set fire time at X, you can get the bookmark within (X-1,X+1) period.
+// This is NOT thread-safe.
+type timeBucketWatchers struct {
+	watchersQueue map[int64][]*cacheWatcher
+	startBucketID int64
```

Comment:
rename `watcherQueue` to `watchersBuckets`? Or is there a difference between queue and bucket here? If so, let's document.

Reply:
i think this should be named as watchersBucket

Reply:
rename it to watchersBuckets